app/models/user.rb

```diff
@@ -26,7 +26,7 @@ class User < ActiveRecord::Base
   validates_uniqueness_of :name # done by devise
   validates_uniqueness_of :email # done by devise
 
-  validates :joinedwithcode, :presence => true, :inclusion => { :in => $codes, :message => "%{value} is not valid" }, :on => :create
```

Comment:
So maybe we can get rid of the $codes global variable then? I have to find it again...